### DISCUSSION TITLE
Is it possible to use the WiFi shield using this?

### DISCUSSION THREAD

**Discussion by NickLarsenNZ** [Rahix/avr-hal#325] [Q&A]
I have an old Arduino Uno (r3) that I'm bringing back to life, and it has a wifi shield.

I can't see any mention of wifi in this library. I do see some other libraries that provide wifi (such as [esp-idf-svc::wifi](https://github.com/esp-rs/esp-idf-svc/blob/master/src/wifi.rs)) but I'm unsure if this can be used with other libs.

**Comment by Rahix** [comment]
Right now, no.  You will need to port the C library to Rust or build FFI bindings for it to make this work.  Both are not easy and will take quite some effort, though...
